cpp/include/cugraph_c/graph_generators.h

```diff
@@ -0,0 +1,258 @@
+ * @return error code
+ */
+cugraph_error_code_t cugraph_generate_rmat_edgelists(
+  const cugraph_resource_handle_t* handle,
+  cugraph_rng_state_t* rng_state,
+  size_t n_edgelists,
+  size_t min_scale,
+  size_t max_scale,
+  size_t edge_factor,
+  cugraph_generator_distribution_t size_distribution,
+  cugraph_generator_distribution_t edge_distribution,
+  bool_t clip_and_flip,
+  cugraph_coo_list_t** result,
+  cugraph_error_t** error);
+
+/**
+ * @brief      Generate edge weights and add to an rmat edge list
+ *
+ * Updates a COO to contain random edge weights
+ *
+ * @param [in]     handle             Handle for accessing resources
+ * @param [in/out] rng_state          State of the random number generator, updated with each call
+ * @param [in/out] coo                Opaque pointer to the coo, weights will be added (overwriting
+ * any existing weights)
+ * @param [in]     dtype              The type of weight to generate (FLOAT32 or FLOAT64), ignored
+ * unless include_weights is true
+ * @param [in]     minimum_weight     Minimum weight value to generate
+ * @param [in]     maximum_weight     Maximum weight value to generate
+ * @param [out]    error              Pointer to an error object storing details of any error.  Will
+ *                                    be populated if error code is not CUGRAPH_SUCCESS
+ */
+cugraph_error_code_t cugraph_generate_edge_weights(const cugraph_resource_handle_t* handle,
+                                                   cugraph_rng_state_t* rng_state,
+                                                   cugraph_coo_t* coo,
+                                                   cugraph_data_type_id_t dtype,
+                                                   double minimum_weight,
+                                                   double maximum_weight,
+                                                   cugraph_error_t** error);
+
+/**
+ * @brief      Add edge ids to an RMAT COO
+ *
+ * Updates a COO to contain edge ids.  Edges will be numbered from 0 to n-1 where n is the number of
+ * edges
+ *
+ * @param [in]     handle             Handle for accessing resources
+ * @param [in/out] coo                Opaque pointer to the coo, weights will be added (overwriting
+ * any existing weights)
+ * @param [in]     multi_gpu          Flag if the COO is being created on multiple GPUs
+ * @param [out]    error              Pointer to an error object storing details of any error.  Will
+ *                                    be populated if error code is not CUGRAPH_SUCCESS
+ */
+cugraph_error_code_t cugraph_generate_edge_ids(const cugraph_resource_handle_t* handle,
+                                               cugraph_coo_t* coo,
+                                               bool_t multi_gpu,
+                                               cugraph_error_t** error);
+
+/**
+ * @brief      Generate random edge types, add them to an RMAT COO
```

Comment:
Same question here: does it have to be an RMAT COO?

Reply:
See above answer